### PR TITLE
Remove dim 1 special case and correct ans, fixes #3235

### LINKED ISSUE #3235: q-expansion download to Sage code bugs
The code to download a CMF q-expansion to Sage has a couple of issues,

- The first is that a_{p^k}'s are not set properly for bad p https://github.com/LMFDB/lmfdb/blob/1ebf496611f83255e4b593975bd68b6d404b7ddd/lmfdb/classical_modular_forms/download.py#L128 (this can be seen on http://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/14/2/a/a/, download the q-expansion code for sage and the q^4 term disappears).
Maybe how does this look as a replacement:
```sage
            '    k = RR(an_list_bound).log(p).floor() + 1',
            '    if p.divides(level):',
            '        euler_factor = [1, -ap, p**(weight - 1) * char_values[p]]',
            '    else:',
            '        euler_factor = [1, -ap]',
            '    foo = (1/PS(euler_factor)).padded_list(k)',
            '    for i in range(1, k):',
            '        an[p**i] = foo[i]',
```

- The second is that https://github.com/LMFDB/lmfdb/blob/1ebf496611f83255e4b593975bd68b6d404b7ddd/lmfdb/classical_modular_forms/download.py#L169 sets all character values to be 1 in dimension 1 but the character could be order 2 as in http://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/23/1/b/a/ (the q^25 term differs in the downloaded q-exp) maybe it just needs to be changed to an order 1 special case?

### PR DESCRIPTION
**Pull request by alexjbest**
See #3235

### CODE REVIEW

File: lmfdb/classical_modular_forms/download.py [hunk 1]
```diff
@@ -165,39 +160,34 @@ def download_qexp(self, label, lang='sage'):
...
             else:
-                explain += c + ' Each a_p is given as a linear combination\n'
-                explain += c + ' of the following basis for the coefficient ring.\n'
-                poly_data = '\n' + c + ' The following line gives the coefficients of\n'
-                poly_data += c + ' the defining polynomial for the coefficient field.\n'
-                poly_data =  self.assign(lang, 'poly_data', hecke_nf['field_poly'], level = 1)
-                if hecke_nf['hecke_ring_power_basis']:
-                    basis_data = '\n' + c + ' The basis for the coefficient ring is just the power basis\n'
-                    basis_data += c + ' in the root of the defining polynomial above.\n'
-                    basis_data = ''
```
Comment (alexjbest) [inline lmfdb/classical_modular_forms/download.py]: One other presumed bug fixed here.

**Comment by roed314** [comment]
Looks good; you just need to update the failing test.

**Comment by alexjbest** [comment]
Great, I was waiting for Travis to tell me what it is but if he doesnt I'll run them on my machine

**Comment by roed314** [comment]
It's just one failure:
```
__________________________ CmfTest.test_download_qexp __________________________
self = <lmfdb.classical_modular_forms.test_cmf.CmfTest testMethod=test_download_qexp>
    def test_download_qexp(self):
        for label, exp in [
                ['11.7.b.a'
                    , '[0, 10, -64]'],
                ['11.2.a.a'
                    , '[-2, -1, 2]'],
                ['21.2.g.a'
                    , '[0, -a - 1, 2*a - 2]'],
                ['59.2.a.a'
                    , '[-a^4 + 7*a^2 + 3*a - 5, a^4 - a^3 - 6*a^2 + 2*a + 3, a^3 - a^2 - 4*a + 3]'],
                ['13.2.e.a'
                    , '[-a - 1, 2*a - 2, a]'],
                ['340.1.ba.b'
                    , '[z, 0, 0]'],
                ['24.3.h.a'
                    , '[-2, 3, 0]'],
                ['24.3.h.c'
                    , '[a, -a^2 - 3, 0]'],
                ]:
            sage_code = self.tc.get('/ModularForm/GL2/Q/holomorphic/download_qexp/%s' % label, follow_redirects=True).data
            assert "make_data" in sage_code
            assert "aps_data" in sage_code
            sage_code += "\n\nout = str(make_data().list()[2:5])\n"
            exec(sage_code, globals())
            global out
>           assert str(out) == exp
E           AssertionError: assert '[0, 10, 64]' == '[0, 10, -64]'
E             - [0, 10, 64]
E             + [0, 10, -64]
E             ?         +
lmfdb/classical_modular_forms/test_cmf.py:559: AssertionError
```

**Comment by alexjbest** [comment]
Ok CI is now happy.

**Comment by roed314** [comment]
LGTM

**Pull request merged**